pytket/docs/changelog.md

```diff
@@ -0,0 +1,1750 @@
+# Changelog
+
+## Unreleased
+
+API changes:
+
+- A number of utility methods and classes pertaining to classical expressions,
+  pass scripting, QASM and Quipper parsing, SPAM mitigation, and symbolic
+  evaluation of circuits have been renamed to indicate they are not intended
+  for external usage.
+
+Fixes:
+
+- Ensure downcasting of `UnitID` to `Qubit` in initial and final maps.
+
+## 2.3.2 (April 2025)
+
+Features:
+
+- Add option to `Circuit.remove_blank_wires` to force removal of all blank
+  classical wires.
+
+Fixes:
+
+- Apply special handling of multi-controlled gates with 0 or 1 controls when
+  adding operations using `UnitID` arguments.
+
+## 2.3.1 (April 2025)
+
+Fixes:
+
+- Fix C++/Python binding of fidelities information passed to `DecomposeTK2`
+  constructor.
+
+## 2.3.0 (April 2025)
+
+API changes:
+
+- Gate construction methods (e.g. {py:meth}`Circuit.X`) have a streamlined API
+  by accepting `Union` types in place of overloaded methods, giving better
+  error messages when mixing {py:class}`UnitID` /{py:class}`Qubit` /
+  {py:class}`Bit` and `int` for gate arguments.
+
+Features:
+
+- Enabling pickling of `Architecture` objects.
+- Added optional inclusive filters to the
+  {py:meth}`~pytket.passes.DecomposeBoxes` pass to only decompose boxes of
+  given types or opgroups.
+- New utility function `circuit_dict_from_pytket1_dict` to convert
+  serializations of pytket 1 circuits containing `ClassicalExpBox` es to
+  serializations of equivalent pytket 2 circuits containing `ClExprOp` s.
+
+Performance:
+
+- Reduced the wall-clock time of `GreedyPauliSimp` by approximately 40% for large circuits.
+- Improved performance of Python/C++ bindings.
+
+## 2.2.0 (April 2025)
+
+Features:
+
+- Add methods in `circuit_library` to convert `CX` and `TK2` to and from
+  `ISWAPMax`, and support for `AutoRebase` targeting `ISWAPMax`.
+
+Fixes:
+
+- Make `ZXGraphlikeOptimisation` pass preserve the circuit's name.
+- Fix handling of bitwise inequality conditions when parsing QASM.
+- Define transform used in `scratch_reg_resize_pass` as as separate function.
+- Enable deserialization of `NormaliseTK2` pass.
+- Don't remove classical wires in `FlattenRelabelRegistersPass`.
+
+## 2.1.0 (March 2025)
+
+Features:
+
+- Add `target_2qb_gate` parameter to `CliffordSimp` pass and `OptimiseCliffords`
+  transform.
+- Add `tk2_to_cx` parameter to `RebaseToCliffordSingles` transform.
```

Comment:
Did you intend to do the myst crossreffing in this PR or is that to be done in a follow-up?

Reply:
No, I figured I would keep this PR as close as possible to just running the converter. Turning on the nitpicky sphinx option showed hundreds of already broken crossrefs that would need to be fixed by hand before we even get to adding new references, both of which would have required this PR to have a full in-depth review rather than just a skim over to check what was done by the automatic tool

Reply:
Makes sense :) Just checking